homeassistant/components/sensor/postnl.py

```diff
@@ -0,0 +1,129 @@
+"""
+Sensor for PostNL packages.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.postnl/
+"""
+from collections import defaultdict
+import logging
+from datetime import timedelta, datetime
+import re
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_USERNAME, CONF_PASSWORD,
+                                 ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['postnl_api==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+CONF_UPDATE_INTERVAL = 'update_interval'
+DOMAIN = 'postnl'
+ICON = 'mdi:package-variant-closed'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=1800)):
+        vol.All(cv.time_period, cv.positive_timedelta),
+})
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the PostNL platform."""
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    name = config.get(CONF_NAME)
+    update_interval = config.get(CONF_UPDATE_INTERVAL)
+
+    from postnl_api import PostNL_API
+
+    try:
+        api = PostNL_API(username, password)
+
+    except Exception:
+        _LOGGER.exception('Wrong Credentials')
```

Comment:
Might be better if this is changed to 'Error when logging in'. It currently will say Wrong Credentials if the PostNL site is down.